javalib/src/main/scala/java/io/ByteArrayInputStream.scala

```diff
@@ -39,6 +39,18 @@ class ByteArrayInputStream(
     }
   }
 
+  def readNBytes(b: Array[Byte], off: Int, len: Int): Int = {
+    val n = read(b, off, len)
+    if (n == -1) 0 else n
+  }
+
+  def readNBytes(len: Int): Array[Byte] = {
```

Comment:
The squirrels running around inside my brain woke me out
of my nap to ask:  What happens if `len` in line 47 is negative?
Does the proper exception `IllegalArgumentException - if length is negative` get thrown?

I said "Don't be silly, no one would actually pass a negative
value. If they did, this code would throw `java.lang.NegativeArraySizeException` and they would be hosed."  Then I stretched, wrote this note, and went off to seek supper.

PS:  I believe that the 3arg variant is OK in this regard. It
        passes the negative value down to a read, which 
        then throws the `IndexOutOfBoundsException`  
        documented for that variant.